### ISSUE TITLE
[Minor autocomplete bug?] Clang declspec properties are lower priority 

### ISSUE THREAD

**Issue by Fernthedev** [clangd/clangd#1767]
`clangd` is able to recognize correctly that a property exists and is valid. However, when trying to autocomplete, it is treated as what I would call a "3rd class citizen" option that is usually resorted to VS Code's least likely useful autocomplete entry. 

My suggestion is specifically: Treat it similar to how C# treats properties prioritization.

C++:
![image](https://github.com/clangd/vscode-clangd/assets/15272073/b6e1a14f-4184-42b7-bd2c-58c379269ac3)

C#:
![image](https://github.com/clangd/vscode-clangd/assets/15272073/6263297d-b34b-446f-9433-0a60b72662a4)

**System information**
Clangd version (from the log, or `clangd --version`):
clangd extension version: 16.0.2
Operating system: Windows 10 VS Code 1.82.2 Stable

**Comment by HighCommander4** [comment]
(Moved to server repo as this is a server-side issue independent of the vscode client)

**Comment by Fernthedev** [comment]
Interesting, because in my clueless digging I found some (but also not many) cases of properties being handled. Example:
https://github.com/llvm/llvm-project/blob/e1a5bb59b91d60c0d87feb78f0e0614589a4c927/clang-tools-extra/clangd/Quality.cpp#L131-L133

However, this may be specific to Obj-C properties, which are tested here: https://github.com/llvm/llvm-project/blob/e1a5bb59b91d60c0d87feb78f0e0614589a4c927/clang-tools-extra/clangd/unittests/HoverTests.cpp#L2921

I couldn't do much on my own to try PR a solution, hopefully someone else can. 👍

**Comment by HighCommander4** [comment]
Yeah, it looks like the ObjC feature and this language extension use two distinct AST node types (`ObjCPropertyDecl` vs. `MSPropertyDecl`). Support for `ObjCPropertyDecl` has been contributed over time (I think by an Apple engineer), but I don't think anyone has worked on `MSPropertyDecl`.
